src/Contract.php

```diff
@@ -187,6 +187,9 @@ public static function rawSearchOptionsToAdd()
                 'joinparams' => [
                     'jointype' => 'itemtype_item'
                 ]
+            ],
+            'condition'          => [
+                'NEWTABLE.entities_id' => $_SESSION['glpiactiveentities'] ?? '0',
```

Comment:
```suggestion
                'NEWTABLE.entities_id' => $_SESSION['glpiactiveentities'] ?? [],
```
'0' is the root entity

Reply:
@cedric-anne, @trasher, if you're OK with my suggestion, can you apply it ? As I don't have the rights to Samuel's repo to do it myself.

Reply:
Not sure it is correct, it will generate an empty `IN` condition, and it will result in an invalid query.

Reply:
Indeed, an empty array can cause issues; but `0` is incorrect also.

`$_SESSION['glpiactiveentities']` is widely used in source code without any fallback value. In facdts, it should always been defined once user is logged. The only "legitimate" case would be from iunit tests (and in that case, setting the value in session, or properly login user seems the better solution).

Reply:
I had set it to 0 by default because during unit tests, a large number of them would not retrieve any value for $_SESSION['glpiactiveentities'], preventing them from passing.

Reply:
Anyway, this condition does not make sense as a link between two items that are not visible from a same context is an invalid link.

Reply:
Tests related issues must be fixed in tests. Using `0` as default value is incorrect. Anyway, as Cédric said, this condition should not exists, so, no problem :)